examples/_web-components-template.html

```diff
@@ -272,38 +272,34 @@ <h2>Modal Dialog example</h2>
       size="wide"
       backdrop-click-exits="true"
     >
-      <div>
-        <p>
-          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan diam vitae metus
-          lacinia, eget tempor purus placerat.
-        </p>
-        <div slot="heading">
-          <h2>This is the modal heading.</h2>
-        </div>
-        <div slot="actions">
-          <form method="dialog">
-            <ds-button type="reset" value="rstBtn" class-name="ds-u-margin-right--1"
-              >Reset</ds-button
-            >
-            <ds-button type="submit" value="sbmBtn">Confirm</ds-button>
-          </form>
-        </div>
+      <div slot="heading">
+        <h2>This is the modal heading.</h2>
+      </div>
+      <p>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan diam vitae metus
+        lacinia, eget tempor purus placerat.
+      </p>
+
+      <div slot="actions">
+        <form method="dialog">
+          <ds-button type="reset" value="rstBtn" class-name="ds-u-margin-right--1">Reset</ds-button>
+          <ds-button type="submit" value="sbmBtn">Confirm</ds-button>
+        </form>
       </div>
     </ds-modal-dialog>
 
     <script>
       document.addEventListener('DOMContentLoaded', (event) => {
         const modal = document.querySelector('ds-modal-dialog');
         const toggleButton = document.getElementById('modal-toggle');
+        const form = modal.querySelector('form[method="dialog"]');
 
-        const openModal = () => {
-          modal?.setAttribute('is-open', 'true');
-        };
+        const openModal = () => modal?.setAttribute('is-open', 'true');
+        const closeModal = () => modal?.setAttribute('is-open', 'false');
 
-        modal?.addEventListener('ds-exit', () => {
-          modal?.setAttribute('is-open', 'false');
-        });
+        modal?.addEventListener('ds-exit', closeModal);
         toggleButton?.addEventListener('click', openModal);
+        form?.addEventListener('submit', closeModal);
```

Comment:
I'm pretty sure it's because the event is being triggered by an element rendered in a slot, which is outside the Shadow DOM where the event is being listened for, so it doesn't bubble up to it.

Reply:
Just read this article which was quite helpful: https://javascript.info/shadow-dom-events

You're exactly correct @pwolfert, I did not know about event retargeting and that it doesn't happen for slotted content!

Reply:
Nice find!